File: problema 14/fortran/problema_14.f90
program compararDiagonales
    implicit none
    integer :: matriz1(12,12), matriz2(12,12)
    integer :: i, j
    logical :: sonIguales
    sonIguales = .true.

    ! Leer las matrices
    print *, "Ingrese los elementos de la primera matriz:"
    do i = 1, 12
        do j = 1, 12
            read *, matriz1(i,j)
        end do
    end do

    print *, "Ingrese los elementos de la segunda matriz:"
    do i = 1, 12
        do j = 1, 12
            read *, matriz2(i,j)
        end do
    end do

    ! Comparar las diagonales principales
    do i = 1, 12
        if (matriz1(i,i) /= matriz2(i,i)) then
            sonIguales = .false.
            exit
        endif
    end do

    if (sonIguales) then
        print *, "Las diagonales principales son iguales."
    else
        print *, "Las diagonales principales no son iguales."
    endif

end program compararDiagonales

File: problema 14/c-languaje/problema_14.c
#include <stdio.h>

int main() {
    int matriz1[12][12], matriz2[12][12];
    int i, j;
    int sonIguales = 1;  // Verdadero

    // Leer las matrices
    printf("Ingrese los elementos de la primera matriz:\n");
    for (i = 0; i < 12; i++) {
        for (j = 0; j < 12; j++) {
            scanf("%d", &matriz1[i][j]);
        }
    }

    printf("Ingrese los elementos de la segunda matriz:\n");
    for (i = 0; i < 12; i++) {
        for (j = 0; j < 12; j++) {
            scanf("%d", &matriz2[i][j]);
        }
    }

    // Comparar las diagonales principales
    for (i = 0; i < 12; i++) {
        if (matriz1[i][i] != matriz2[i][i]) {
            sonIguales = 0;  // Falso
            break;
        }
    }

    if (sonIguales) {
        printf("Las diagonales principales son iguales.\n");
    } else {
        printf("Las diagonales principales no son iguales.\n");
    }

    return 0;
}